src/core/profile/resolve-profile-ast.ts

```diff
@@ -81,53 +81,44 @@ export async function resolveProfileAst({
   ) {
     throw versionMismatchError(profileSettings.version, version);
   }
-  let filepath: string;
+  let filepath: string, astPath: string;
 
   let resolvedVersion: string;
   // TODO: do we want to check `file` if we have version from getProfile?
   if (superJson !== undefined && profileSettings !== undefined) {
     if ('file' in profileSettings) {
       filepath = fileSystem.path.resolve(
-        config.superfacePath,
+        fileSystem.path.dirname(config.superfacePath),
         profileSettings.file
       );
-      logFunction?.('Reading possible profile file: %s', filepath);
-      // check extensions
+      // if we find source file we assume compiled file next to it
       if (filepath.endsWith(EXTENSIONS.profile.source)) {
+        astPath = filepath.replace(
+          EXTENSIONS.profile.source,
+          EXTENSIONS.profile.build
+        );
+
+        if (!(await fileSystem.exists(astPath))) {
+          throw sourceFileExtensionFoundError(EXTENSIONS.profile.source);
+        }
```

Comment:
issues: if I am reading it right, it checks for presense of ast, but throws error for missing source document. Should it be?

```suggestion
        if (!(await fileSystem.exists(astPath))) {
          throw sourceFileExtensionFoundError(EXTENSIONS.profile.build);
        }
```

Reply:
This one is correct - this informs user that .supr (source) file must be compiled.

This case will happen when you are migrating from SDK 1.x to 2.0 with locally linked files

Reply:
🤦 sorry didn't check what is inside `sourceFileExtensionFoundError`. It is alright.